sidecar/internal/service/reclaimspace.go

```diff
@@ -61,17 +69,45 @@ func (rs *ReclaimSpaceServer) ControllerReclaimSpace(
 	pvName := req.GetPvName()
 	klog.Info(pvName)
 
-	// TODO: get following params using kubeClient and pvName
+	pv, err := rs.kubeClient.CoreV1().PersistentVolumes().Get(ctx, pvName, metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("Failed to get pv: %v", err)
+		return nil, status.Errorf(codes.InvalidArgument, "failed to get pv %q", pvName)
+	}
+
+	if pv.Spec.CSI == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "pv %q is not a CSI volume", pvName)
+	}
 
+	volID := pv.Spec.CSI.VolumeHandle
+	volAttributes := pv.Spec.CSI.VolumeAttributes
+	var deviceSecret map[string]string
 	csiReq := &csiReclaimSpace.ControllerReclaimSpaceRequest{
-		VolumeId:   "",
-		Parameters: map[string]string{},
-		Secrets:    map[string]string{},
+		VolumeId:   volID,
+		Parameters: volAttributes,
+		Secrets:    deviceSecret,
```

Comment:
`Secrets` do not need to be set here, they are optional. Just drop the whole `Secrets:    deviceSecret,` line.